### PR TITLE
[Environments] - Dark mode support and RTK migration

### PR DESCRIPTION
**Pull request by senali-d**
**Notes for Reviewers**

This PR fixes #

- Dark mode support for environments
- Migrate api to RTK
- Scroll loading for transfer list component
- Fix caching issues

Screenshot 2023-12-25 at 6 58 58 PM

Screenshot 2023-12-25 at 6 59 21 PM

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — COMMENTED** [review]
The existing light mode coloring can be left as is, I think. The dark mode only needs the white card to be changed to a shade of gray, I believe. 

@Rexford74, we have pre-established colors in-use in Meshery UI for similar cards in dark mode. I assume that these lineup with the proposed colors in Sistent. Is that about right?

**Comment by ritiksaxena124** [comment]
@senali-d we already have a mockup for the dark mode of environments card in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI?type=design&node-id=24380%3A25009&mode=design&t=VPJrl1doCmoZC5EO-1). I mentioned you on that you might have missed it previously.

**Comment by Rexford74** [comment]
> The existing light mode coloring can be left as is, I think. The dark mode only needs the white card to be changed to a shade of gray, I believe.
> 
> @Rexford74, we have pre-established colors in-use in Meshery UI for similar cards in dark mode. I assume that these lineup with the proposed colors in Sistent. Is that about right?

Well, so far, just Layer5's colors have been defined and none of the colors for the different products: Meshery, Nighthawk, etc.
This is not an issue however as any colors for individual products of Layer5 can be added both now and in the future as the need arises. If it is possible, please I will like a confirmation of what the primary color for Meshery is; Charcoal or Teal Blue. This will help to provide an accurate tonal palette and incorporate the addition to the designed Sisten drafts.

**Comment by ritiksaxena124** [comment]
> > The existing light mode coloring can be left as is, I think. The dark mode only needs the white card to be changed to a shade of gray, I believe.
> > @Rexford74, we have pre-established colors in-use in Meshery UI for similar cards in dark mode. I assume that these lineup with the proposed colors in Sistent. Is that about right?
> 
> Well, so far, just Layer5's colors have been defined and none of the colors for the different products: Meshery, Nighthawk, etc. This is not an issue however as any colors for individual products of Layer5 can be added both now and in the future as the need arises. If it is possible, please I will like a confirmation of what the primary color for Meshery is; Charcoal or Teal Blue. This will help to provide an accurate tonal palette and incorporate the addition to the designed Sisten drafts.

Teal is not a primary color @Rexford74

File: ui/themes/app.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ export var darkTheme = createTheme({
       disableButtonBg: '#666666',
       disableButton: '#9d9d9d',
       listItemHover: 'rgb(0, 187, 166, 0.5)',
-      white: 'rgb(255, 255, 255, 0.1)',
+      white: '#202020',
```
Comment (Yashsharma1911) [inline ui/themes/app.js:120]: // FYI @ritiksaxena124

Comment (Yashsharma1911) [reply ui/themes/app.js:120]: @senali-d have you tested this theme change in MeshMap too?

Comment (senali-d) [reply ui/themes/app.js:120]: This color define specifically for environments in my previous PR. I think this was not use anywhere other than this page.

Comment (senali-d) [reply ui/themes/app.js:120]: #9647

**Comment by senali-d** [comment]
Suggestion for APIs: 

1. Allocated connections count return from the get environments API: `api/environments`
2. Bulk allocation of connections API

- When we go ahead with the scroll loading we need a separate api for get the assignable connections to a specific environment.

// @theBeginner86 @aabidsofi19

File: ui/themes/app.js (hunk 1 above)
Comment (Yashsharma1911) [reply ui/themes/app.js:120]: Ohh make sense thanks for confirming

**Comment by senali-d** [comment]
I have temporary disable scroll loading of transfer list items (connections) to move forward...

**Pull request merged**